include/xtensor/xexception.hpp

```diff
@@ -92,6 +92,26 @@ namespace xt
         return m_message.c_str();
     }
 
+    /*****************************
+     * not_implemented_exception *
+     *****************************/
+
+    class not_implemented_exception : public std::logic_error
```

Comment:
I think this can be removed since you use static_assert for non implemented stuff.